src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/2.sql

```diff
@@ -871,10 +871,12 @@ AS
          Watermark BIGINT)
 
     DECLARE @studyKey BIGINT
+    DECLARE @seriesKey BIGINT
+    DECLARE @instanceKey BIGINT
     DECLARE @deletedDate DATETIME2 = SYSUTCDATETIME()
 
-    -- Get the study PK
-    SELECT  @studyKey = StudyKey
+    -- Get the study, series and instance PK
+    SELECT  @studyKey = StudyKey, @seriesKey = SeriesKey, @instanceKey = InstanceKey
```

Comment:
does this work when deleting Study with multiple series or instance? you cannot have a single series or instance

Reply:
updated this to work when instanceUid and seriesUid are not provided.